noxfile.py

```diff
@@ -684,220 +681,13 @@ def notebook(session: nox.Session):
-    notebooks_reg = {
-        os.path.join("notebooks/location", nb): regions
-        for nb, regions in notebooks_reg.items()
-    }
-
-    # The pytest --nbmake exits silently with "no tests ran" message if
-    # one of the notebook paths supplied does not exist. Let's make sure that
-    # each path exists.
-    for nb in notebooks + list(notebooks_reg):
-        assert os.path.exists(nb), nb
-
-    # TODO(shobs): For some reason --retries arg masks exceptions occurred in
-    # notebook failures, and shows unhelpful INTERNALERROR. Investigate that
-    # and enable retries if we can find a way to surface the real exception
-    # bacause the notebook is running against real GCP and something may fail
-    # due to transient issues.
-    pytest_command = [
-        "py.test",
-        "--nbmake",
-        "--nbmake-timeout=900",  # 15 minutes
-    ]
-
-    try:
-        # Populate notebook parameters and make a backup so that the notebooks
-        # are runnable.
-        session.run(
-            "python",
-            CURRENT_DIRECTORY / "scripts" / "notebooks_fill_params.py",
-            *notebooks,
-        )
-
-        # Run notebooks in parallel session.run's, since each notebook
-        # takes an environment variable for performance logging
-        processes = []
-        for notebook in notebooks:
-            process = Process(
-                target=session.run,
-                args=(*pytest_command, notebook),
-                kwargs={"env": {LOGGING_NAME_ENV_VAR: os.path.basename(notebook)}},
-            )
-            process.start()
-            processes.append(process)
-
-        for process in processes:
-            process.join()
-
-    finally:
-        # Prevent our notebook changes from getting checked in to git
-        # accidentally.
-        session.run(
-            "python",
-            CURRENT_DIRECTORY / "scripts" / "notebooks_restore_from_backup.py",
-            *notebooks,
-        )
-
-    # Additionally run regionalized notebooks in parallel session.run's.
-    # Each notebook takes a different region via env param.
-    processes = []
-    for notebook, regions in notebooks_reg.items():
-        for region in regions:
-            process = Process(
-                target=session.run,
-                args=(*pytest_command, notebook),
-                kwargs={
-                    "env": {
-                        "BIGQUERY_LOCATION": region,
-                        LOGGING_NAME_ENV_VAR: os.path.basename(notebook),
-                    }
-                },
-            )
-            process.start()
-            processes.append(process)
-
-    for process in processes:
-        process.join()
-
-    # when the environment variable is set as it is above,
-    # notebooks output a .bytesprocessed and .slotmillis report
-    # collect those reports and print a summary
-    _print_performance_report("notebooks/")
+    session.run("python", "scripts/publish_benchmark_result.py", "--notebook")
```

Comment:
Rather than totally remove this, let's have a `session.run` per notebook. We can make the run and publish benchmark script take a file path as an argument.

Otherwise we lose out on the stdin/stdout and error code benefits of using nox (see my comments about Popen below).

Reply:
Logic updated. I assume set env_var in kokoro cfg file and read them when publish can work?